docs: More legible font colour in dark mode

cdrage opened the pull request:
Website: More legible font colour in dark mode

### What does this PR do?

* Updates the website with "gray-300" font colour in dark mode, as the
  previous colour was not very legible to read.

### Screenshot/screencast of this PR

<!-- Please include a screenshot or a screencast explaining what is doing this PR -->

See below difference:
![Screen Shot 2022-10-11 at 1 21 49 PM](https://user-images.githubusercontent.com/6422176/195158281-2b0057df-2b64-4a21-aac2-fb9602089f61.png)

vs

![Screen Shot 2022-10-11 at 1 18 35 PM](https://user-images.githubusercontent.com/6422176/195158296-bd4cc5ec-741d-4000-8f70-d438834961c0.png)



### What issues does this PR fix or reference?

<!-- Please include any related issue from Podman Desktop repository (or from another issue tracker).
-->

N/A (let me know if I need to open an issue)! It's a quickfix for the
website.

### How to test this PR?

`yarn run docusaurus start`

<!-- Please explain steps to reproduce -->

slemeur commented on website/src/pages/index.tsx:
```diff
@@ -223,13 +221,15 @@ function Extensibility() {
           <div className="flex flex-col items-center">
             <div className="text-left my-4">
-              <p className="-ml-5">Current Podman Desktop plug-ins: Podman, Docker, Lima and CRC/OpenShift Local.</p>
+              <p className="-ml-5 text-base">Current Podman Desktop plug-ins: Podman, Docker, Lima and CRC/OpenShift Local.</p>
```
maybe, instead of defining a new class, it should be the "default" color for dark theme which should be changed?

cdrage replied:
I believe this is actually *missing* the class! Most <p> have `text-base` at the class.

slemeur commented on website/src/pages/index.tsx:
```diff
@@ -117,7 +117,7 @@ function Hero() {
             Containers and Kubernetes for application developers
           </h1>
-          <p className="text-base text-gray-700 dark:text-gray-500 md:text-lg">
+          <p className="text-base text-gray-700 md:text-lg">
```
`text-gray-700` is supposed to be giving the color to the paragraph. 
Here, by overriding the class, you are defining the color in another place and it's not following the "tailwind" practices.

I think it would be easier to switch the color of the paragraph to `text-gray-100` or something like that. 
You can check the colors provided out of the box by tailwind here: https://tailwindcss.com/docs/text-color

slemeur reviewed (changes requested):
Good catch and nice improvements.

cdrage reviewed (commented):
I've gone ahead and updated the PR! I didn't actually need the `dark:` class.

Looks like the custom css will accurate fix all the issues with the legibility.

Still learning tailwindcss, so hopefully this looks good!

cdrage commented:
Linter still failing, going to update this PR 👍

benoitf commented on website/src/css/custom.css:
```diff
@@ -107,3 +107,8 @@ svg.lightToggleIcon_node_modules-\@docusaurus-theme-classic-lib-next-theme-Color
 li.footer__item a svg {
   display: inline;
 }
+
```
We should not customize it there

cdrage replied:
Where would be the best way to customize? looking through tailwindcss docs I found that you can use `[data-theme='dark']` to set it in a custom.css

Still learning, so I apologize!

benoitf replied:
I think we're following the 'we set the colors where we need it' rule

so if you want to use one color for the text, you apply 'text-theme-level' for white theme and 'dark:text-theme-level' for dark theme.

so global override, it's a per element customization

cdrage replied:
Got it, so each `p`, I'll add a `dark:text-gray-100` instead. Although it will be more line changes, at least it's the tailwindcss way.

benoitf replied:
what you can do, is a new component like <Subtitle> and inside you apply dark:text-gray-100 and everytime you want a "subtitle" you reuse that component instead of <p>

benoitf replied:
leveraging that there is react or svelte under the hood, so bring new component/helpers

cdrage commented:
Updated!

I've:
* Made it more "tailwindcss"-like, not using a custom.css file
* Fixed some of the classes that had font set incorrectly or didn't need it (font colour set by the section)

benoitf commented on website/src/pages/index.tsx:
```diff
@@ -223,13 +223,19 @@ function Extensibility() {
-          <p className="leading-relaxed">JavaScript extensions can contribute new behaviour</p>
-          <p className="leading-relaxed">Reuse existing extensions directly in Podman Desktop</p>
+          <p className="leading-relaxed text-base">
+            Container engines are plugged through extension points.
+            <br />
+            JavaScript extensions can contribute new behaviour
+            <br />
+            Reuse existing extensions directly in Podman Desktop
+          </p>
```
Here it looks like more a list, so you could use `<ul><li>` (and by default there is no bullet, it's up to you to choose)
https://tailwindcss.com/docs/list-style-type

benoitf commented on website/src/pages/index.tsx:
```diff
@@ -187,18 +187,18 @@ function KeepUpToDate() {
-            </p>
-            <p className="leading-relaxed">Check for updates and get notified about new changes.</p>
-            <p className="leading-relaxed">
+              <br />
```
br works but probably <p>text</p> or <div>text</div> is more accurate

benoitf reviewed (approved):
reported few minor fixes

thanks 👍

benoitf commented:
@slemeur is it ok now for you ?

slemeur commented:
LGTM 👍
